test/python/circuit/test_circuit_properties.py

```diff
@@ -547,6 +547,37 @@ def test_circuit_unitary_factors4(self):
         qc.measure(q[3], c[0])
         self.assertEqual(qc.num_unitary_factors(), 5)
 
+    def test_n_qubits_qubitless_circuit(self):
+        """Check output in absence of qubits
+        """
+        q_reg = QuantumRegister(0)
+        c_reg = ClassicalRegister(3)
+        circ = QuantumCircuit(q_reg, c_reg)
+        self.assertEqual(circ.n_qubits, 0)
```

Comment:
These tests are failing because you are not calling the method here, as you do not have the brackets `circ.n_qubits()`. If you want to be able to access it in this way you need to add the annotation `@property` above the declaration of the method, or you can update all the calls to be `circ.n_qubits()`. Personally I think the `@property` makes more sense.

Reply:
In the commit "tests moved to circuit properties test file", I did use the brackets but that commit failed the tests. So can you give me an example of how the annotation case would work?

Reply:
Check out the [node_counter](https://github.com/Qiskit/qiskit-terra/blob/b970a2179662b2fc4abfe0a2e8fbf61d8f83f6ad/qiskit/dagcircuit/dagcircuit.py#L97) property for an example. All it involves is putting `@property` above the method definition.

Reply:
I have just had a look, those tests failed because you cannot declare a register of 0 size, so that test needs to be changed. You can [run all the tests locally](https://github.com/Qiskit/qiskit-terra/blob/master/CONTRIBUTING.md) as well to see if your changes work by running `make test` in the terminal.